libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/GutenbergEditorFragment.java

```diff
@@ -0,0 +1,538 @@
+        super.onCreate(savedInstanceState);
+
+        mWPAndroidGlueCode.onCreate(getContext());
+
+        ProfilingUtils.start("Visual Editor Startup");
+        ProfilingUtils.split("EditorFragment.onCreate");
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_gutenberg_editor, container, false);
+
+        mTitle = view.findViewById(R.id.title);
+        mWPAndroidGlueCode.onCreateView(
+                view.findViewById(R.id.gutenberg),
+                new OnMediaLibraryButtonListener() {
+                    @Override public void onMediaLibraryButtonClick() {
+                        onToolbarMediaButtonClicked();
+                    }
+                },
+                getActivity().getApplication(),
+                BuildConfig.DEBUG,
+                BuildConfig.BUILD_GUTENBERG_FROM_SOURCE);
+        mSource = view.findViewById(R.id.source);
+
+        mTitle.addTextChangedListener(mTextWatcher);
+
+        // request dependency injection. Do this after setting min/max dimensions
+        if (getActivity() instanceof EditorFragmentActivity) {
+            ((EditorFragmentActivity) getActivity()).initializeEditorFragment();
+        }
+
+        mTitle.setOnTouchListener(this);
+        mSource.setOnTouchListener(this);
+
+        mTitle.setOnImeBackListener(new OnImeBackListener() {
+            public void onImeBack() {
+                showActionBarIfNeeded();
+            }
+        });
+
+        mTitle.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                for (int i = s.length(); i > 0; i--) {
+                    if (s.subSequence(i - 1, i).toString().equals("\n")) {
+                        s.replace(i - 1, i, " ");
+                    }
+                }
+            }
+        });
+
+        // We need to intercept the "Enter" key on the title field, and replace it with a space instead
+        mSource.setHint("<p>" + getString(R.string.editor_content_hint) + "</p>");
+
+        mFormattingToolbar = view.findViewById(R.id.formatting_toolbar);
+        mFormattingToolbar.setExpanded(mIsToolbarExpanded);
+
+        mTitle.setOnFocusChangeListener(
+                new View.OnFocusChangeListener() {
+                    @Override
+                    public void onFocusChange(View view, boolean hasFocus) {
+                        mFormattingToolbar.enableFormatButtons(!hasFocus);
+                    }
+                }
+        );
+
+        setHasOptionsMenu(true);
+
+        mInvalidateOptionsHandler = new Handler();
```

Comment:
We can do this initialization during the declaration since it doesn't have any extra benefit to do it during `onCreateView` since it'll still block the UI. A better approach would be to make it lazy through a method (or `lazy` modifier in Kotlin 😍 ) so it's not done until it's necessary.